fscollections/models.py

```diff
@@ -0,0 +1,144 @@
+    public = models.BooleanField(default=False)
+    is_default_collection = models.BooleanField(default=False)
+    #NOTE: Don't fear migrations, you're just testing
+    #subcolletion_path = sth with tagn and routing folders for downloads
+    #follow relation for users and collections (intersted but not owner nor contributor)
+
+    def __str__(self):
+        return f"{self.name}"
+    
+    def get_attribution(self, sound_qs=None):
+        #If no queryset of sounds is provided, take it from the bookmark category
+        if sound_qs is None:
+            collection_sounds = CollectionSound.objects.filter(collection=self).values("sound_id")
+            sound_qs = Sound.objects.filter(id__in=collection_sounds, processing_state="OK", moderation_state="OK").select_related('user','license')
+        
+        users = User.objects.filter(sounds__in=sound_qs).distinct()
+        # Generate text file with license info
+        licenses = License.objects.filter(sound__id__in=sound_qs).distinct()
+        attribution = render_to_string(("sounds/multiple_sounds_attribution.txt"),
+            dict(type="Collection",
+                users=users,
+                object=self,
+                licenses=licenses,
+                sound_list=sound_qs))
+        return attribution
+
+    @property
+    def download_filename(self):
+        name_slug = slugify(self.name)
+        username_slug = slugify(self.user.username)
+        return "%d__%s__%s.zip" % (self.id, username_slug, name_slug)
+    
+    def get_total_collection_sounds_length(self):
+        result = self.sounds.aggregate(total_duration=Sum('duration'))
+        return result['total_duration'] or 0
+    
+    def save(self, *args, **kwargs):   
+        self.num_sounds = CollectionSound.objects.filter(collection=self).count()
+        if self.num_sounds > 0:
+            # this need to be reviewed, featured_sound feature is not fully developed
+            csound = CollectionSound.objects.filter(collection=self).first()
+            csound.featured_sound = True
+            csound.save()
+        super().save(*args, **kwargs)
+        
+
+class CollectionSound(models.Model):
+   #this model relates collections and sounds
+   user = models.ForeignKey(User, on_delete=models.CASCADE) 
+   sound = models.ForeignKey(Sound, on_delete=models.CASCADE)
+   collection = models.ForeignKey(Collection, related_name='collectionsound', on_delete=models.CASCADE)
+   featured_sound = models.BooleanField(default=False)
+   created = models.DateTimeField(db_index=True, auto_now_add=True)
+   
+   STATUS_CHOICES = (
+        ("PE", 'Pending'),
+        ("OK", 'Accepted'),
+        ("RE", 'Refused'),
+    )
+   status = models.CharField(db_index=True, max_length=2, choices=STATUS_CHOICES, default="PE")
+   #sound won't be added to collection until maintainers approve the sound
+
+@receiver(post_save, sender=CollectionSound)
+def update_collection_num_sounds(**kwargs):
```

Comment:
It'd be better to have the explicit list of parameters to these signals: https://docs.djangoproject.com/en/5.2/ref/signals/#django.db.models.signals.pre_save
```suggestion
def update_collection_num_sounds(sender, instance, raw, using, update_fields):
```

Then you don't need to set isntance to False in the next line (this will already be None if you're in this situation)